singlestoredb/management/job.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python
+"""SingleStoreDB Cloud Scheduled Notebook Job."""
+import datetime
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from enum import Enum
+
+from .utils import to_datetime
+from .manager import Manager
+
+
+
+class TargetType(Enum):
+    WORKSPACE = "Workspace"
+    CLUSTER = "Cluster"
+    VIRTUAL_WORKSPACE = "VirtualWorkspace"
+
+
+class Status(Enum):
+    UNKNOWN = "Unknown"
+    SCHEDULED = "Scheduled"
+    RUNNING = "Running"
+    COMPLETED = "Completed"
+    FAILED = "Failed"
+    ERROR = "Error"
+    CANCELED = "Canceled"
+
+
+class ExecutionMetadata(object):
+
+    avg_duration_in_seconds: int
+    max_duration_in_seconds: int
+    status: Status
+    count: int
+
+    def __init__(
+        self,
+        avg_duration_in_seconds: int,
+        max_duration_in_seconds: int,
+        status: Status,
+        count: int,
+    ):
+        self.avg_duration_in_seconds = avg_duration_in_seconds
+        self.max_duration_in_seconds = max_duration_in_seconds
+        self.status = status
+        self.count = count
+
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> 'ExecutionMetadata':
+        """
+        Construct an ExecutionMetadata from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`ExecutionMetadata`
+
+        """
+        out = cls(
+            avg_duration_in_seconds=int(obj['avgDurationInSeconds']),
+            max_duration_in_seconds=int(obj['maxDurationInSeconds']),
+            status=obj['status'],
+            count=int(obj['count']),
+        )
+
+        return out
+
+
+class ExecutionConfig(object):
+
+    create_snapshot: Optional[bool]
+    max_allowed_execution_duration_in_minutes: Optional[int]
+    notebook_path: Optional[str]
+
+    def __init__(
+        self,
+        create_snapshot: Optional[bool],
+        max_allowed_execution_duration_in_minutes: Optional[int],
+        notebook_path: Optional[str],
+    ):
+        self.create_snapshot = create_snapshot
+        self.max_allowed_execution_duration_in_minutes = max_allowed_execution_duration_in_minutes
+        self.notebook_path = notebook_path
+
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> 'ExecutionConfig':
+        """
+        Construct an ExecutionConfig from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`ExecutionConfig`
+
+        """
+        out = cls(
+            create_snapshot=bool(obj.get('createSnapshot')),
```

Comment:
For fields that might be missing, `obj.get` will return `None`. Should the output be `None` or a `bool` in this case? If it should be a `bool`, you should use `bool(obj.get('createSnapshot', False))`. And for ints and strs, an appropriate default should be applied. If the resulting value should be a `None`, you'll need to do something like `bool(obj['createSnapshot']) if 'createSnapshot' in obj else None`.